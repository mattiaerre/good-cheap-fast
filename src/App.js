import React, { Component } from 'react';
import Switch from 'react-ios-switch';
import './App.css';
import setToggles from './setToggles';

class App extends Component {
  state = {
    good: false,
    cheap: false,
    fast: false
  };

  render() {
    const { good, cheap, fast } = this.state;

    return (
      <section>
        <h2>Good/Cheap/Fast - pick two</h2>
        <form>
          <label>
            Good
            <Switch
              checked={good}
              className="label__switch"
              onChange={() => {
                this.setState(setToggles('good', this.state));
              }}
            />
          </label>
          <label>
            Cheap
            <Switch
              checked={cheap}
              className="label__switch"
              onChange={() => {
                this.setState(setToggles('cheap', this.state));
              }}
            />
          </label>
          <label>
            Fast
            <Switch
              checked={fast}
              className="label__switch"
              onChange={() => {
                this.setState(setToggles('fast', this.state));
              }}
            />
          </label>
        </form>
        <h3>What are your priorities?</h3>
        <ul>
          <li>Cheap + fast = lower quality work</li>
          <li>Fast + good = expensive</li>
          <li>Good + cheap = not happening anytime soon</li>
        </ul>
      </section>
    );
  }
}

export default App;
