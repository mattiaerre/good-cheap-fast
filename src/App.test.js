import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

test('match snapshot', () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('click good', () => {
  const wrapper = shallow(<App />);
  const good = wrapper.find('.label__switch').first();
  expect(wrapper.state().good).toBe(false);

  good.simulate('change');

  expect(wrapper.state().good).toBe(true);
});

test('click cheap', () => {
  const wrapper = shallow(<App />);
  const good = wrapper.find('.label__switch').at(1);
  expect(wrapper.state().cheap).toBe(false);

  good.simulate('change');

  expect(wrapper.state().cheap).toBe(true);
});

test('click fast', () => {
  const wrapper = shallow(<App />);
  const good = wrapper.find('.label__switch').last();
  expect(wrapper.state().fast).toBe(false);

  good.simulate('change');

  expect(wrapper.state().fast).toBe(true);
});
