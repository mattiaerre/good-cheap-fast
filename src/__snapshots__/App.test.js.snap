// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<section>
  <h2>
    Good/Cheap/Fast - pick two
  </h2>
  <form>
    <label>
      Good
      <span
        className="label__switch"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "border": "1px solid rgba(223, 223, 223, 1)",
            "borderRadius": 15,
            "boxShadow": "inset 0 0 0 0px rgba(223, 223, 223, 1)",
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": 30,
            "opacity": 1,
            "position": "relative",
            "transition": "0.2s",
            "userSelect": "none",
            "width": 50,
          }
        }
      >
        <span
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
              "borderRadius": "100%",
              "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.4)",
              "cursor": "grab",
              "display": "inline-block",
              "height": 28,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transition": "0.2s",
              "width": 28,
            }
          }
        />
        <input
          checked={false}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="checkbox"
        />
      </span>
    </label>
    <label>
      Cheap
      <span
        className="label__switch"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "border": "1px solid rgba(223, 223, 223, 1)",
            "borderRadius": 15,
            "boxShadow": "inset 0 0 0 0px rgba(223, 223, 223, 1)",
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": 30,
            "opacity": 1,
            "position": "relative",
            "transition": "0.2s",
            "userSelect": "none",
            "width": 50,
          }
        }
      >
        <span
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
              "borderRadius": "100%",
              "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.4)",
              "cursor": "grab",
              "display": "inline-block",
              "height": 28,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transition": "0.2s",
              "width": 28,
            }
          }
        />
        <input
          checked={false}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="checkbox"
        />
      </span>
    </label>
    <label>
      Fast
      <span
        className="label__switch"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "border": "1px solid rgba(223, 223, 223, 1)",
            "borderRadius": 15,
            "boxShadow": "inset 0 0 0 0px rgba(223, 223, 223, 1)",
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": 30,
            "opacity": 1,
            "position": "relative",
            "transition": "0.2s",
            "userSelect": "none",
            "width": 50,
          }
        }
      >
        <span
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(255,255,255)",
              "borderRadius": "100%",
              "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.4)",
              "cursor": "grab",
              "display": "inline-block",
              "height": 28,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transition": "0.2s",
              "width": 28,
            }
          }
        />
        <input
          checked={false}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="checkbox"
        />
      </span>
    </label>
  </form>
  <h3>
    What are your priorities?
  </h3>
  <ul>
    <li>
      Cheap + fast = lower quality work
    </li>
    <li>
      Fast + good = expensive
    </li>
    <li>
      Good + cheap = not happening anytime soon
    </li>
  </ul>
</section>
`;
